beacon-chain/db/kv/blocks.go

```diff
@@ -213,11 +215,9 @@ func (s *Store) BlockRootsBySlot(ctx context.Context, slot types.Slot) (bool, []
 	defer span.End()
 	blockRoots := make([][32]byte, 0)
 	err := s.db.View(func(tx *bolt.Tx) error {
-		keys := blockRootsBySlot(ctx, tx, slot)
-		for i := 0; i < len(keys); i++ {
-			blockRoots = append(blockRoots, bytesutil.ToBytes32(keys[i]))
-		}
```

Comment:
I'm not sure why this code is making a new list to copy the return from `blockRootsBySlot`, but I'm pretty sure it isn't necessary.

Reply:
Hey, so there is a reason we usually copy bytes rather than reference them directly. 
https://github.com/prysmaticlabs/prysm/issues/3310
https://github.com/boltdb/bolt/issues/204#issuecomment-46637770

I am not sure if the roots returned by `blockRootsBySlot` are actual keys from the bucket rather than a copy. If it is a copy then we should be fine removing the copy here.